### PR TITLE
Update python dependencies and simplify some of the logic

### PR DESCRIPTION
**Pull request by maylonpedroso**
- Use drf Response class for views responses
- Get token from the request on scope validation
- Use PyJWTClient to get the token validation public key

### CODE REVIEW

File: 01-Authorization/auth0authorization/utils.py [hunk 1]
```diff
@@ -14,16 +15,12 @@ def jwt_get_username_from_payload_handler(payload):
...
-        if jwk['kid'] == header['kid']:
-            public_key = jwt.algorithms.RSAAlgorithm.from_jwk(json.dumps(jwk))
-
-    if public_key is None:
-        raise Exception('Public key not found.')
+    key_client = jwt.PyJWKClient('https://{}/.well-known/jwks.json'.format(settings.AUTH0_DOMAIN))
+    try:
+        public_key = key_client.get_signing_key(header['kid']).key
+    except (jwt.PyJWKClientError, URLError):
+        raise AuthenticationFailed('Public key not found.')
```
Comment (maylonpedroso) [inline 01-Authorization/auth0authorization/utils.py:22]: PyJWT v2 simplifies the retrieval of the public key by adding a PyJWTClient

File: 01-Authorization/auth0authorization/views.py [hunk 2]
```diff
@@ -25,33 +13,29 @@ def requires_scope(required_scope):
     """
     def require_scope(f):
         @wraps(f)
-        def decorated(*args, **kwargs):
-            token = get_token_auth_header(args[0])
-            decoded = jwt.decode(token, verify=False)
+        def decorated(request, *args, **kwargs):
+            decoded = jwt.decode(request.auth, options={'verify_signature': False})
```
Comment (maylonpedroso) [inline 01-Authorization/auth0authorization/views.py:17]: `JSONWebTokenAuthentication` returns the token as part of the authentication tuple: `request.auth` holds the token after successful authentication.

File: 01-Authorization/auth0authorization/utils.py [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
-from django.contrib.auth import authenticate
 import jwt
-import requests
+
+from django.contrib.auth import authenticate
+from django.conf import settings
+from rest_framework.exceptions import AuthenticationFailed
 
 
 def jwt_get_username_from_payload_handler(payload):
```
Comment (maylonpedroso) [inline 01-Authorization/auth0authorization/utils.py:10]: Calling the `authenticate` backend to insert the user in the DB in this method as a side-effect of getting the username from the token payload is confusing.

**Comment by adamjmcgrath** [comment]
Hey @maylonpedroso - thanks for your contribution and apologies for the delay in our response.

We're going a different way with our Django sample (see #36), so are going to close this.

**Pull request closed without merging**